news/three-months-at-the-rust-foundation.md

```diff
@@ -0,0 +1,83 @@
+tags:
+  - foundation
+layout: layouts/news.njk
+---
+
+In May 2022, I received an exciting email from the Rust Foundation:
+
+> […] you have been provisionally selected to be one of our first Rust Foundation Fellows
+
+Fast-forward one year, and that fellowship grant has transformed into a full-time role at the Rust Foundation. My primary focus now revolves around the maintenance and feature development for [crates.io](https://crates.io), the official Rust package registry.
+
+As I reflect on my first three months in this position, I want to take this opportunity to highlight some of the significant milestones I have achieved:
+
+## Improved API tokens
+
+One of the most noticeable achievements during this period has been the substantial improvements made to API tokens on crates.io.
+
+With the introduction of token scopes our users can now restrict their API tokens to specific crates and operations, for instance only allowing a token to publish new versions of a specific crate.
+
+In addition, I implemented optional token expiration, providing users with the ability to configure API tokens to automatically expire at a designated time.
+
+To delve into more details, please refer to the [Rust Blog](https://blog.rust-lang.org/2023/06/23/improved-api-tokens-for-crates-io.html).
+
+## The rust-version field
+
+Crate authors have long been able to specify the minimum supported Rust version for their crate using the `rust-version` field in the `Cargo.toml` file. Previously, `cargo` had to read the `Cargo.toml` file of the downloaded crates to determine potential compatibility issues with the Rust version being used.
+
+Working in collaboration with [Cassaundra Smith](https://github.com/cassaundra) from [Futurewei](https://www.futurewei.com), we implemented a solution that reads the `rust-version` field from the `Cargo.toml` file upon crate upload. The content is then stored in our database and the crate index, enabling `cargo` to eventually utilize it for dependency resolution purposes. To ensure comprehensive support, I also backfilled our database with this field's information for previously uploaded crate versions.
+
+## Resolving Data Inconsistencies
+
+As expected, crates.io has encountered its fair share of bugs and inconsistencies since its inception in 2014. Over the past few months, we have been diligently addressing these issues, focusing on resolving inconsistencies between the database, crate index, and crate storage on S3.
+
+This ranged from versions that were yanked in the database, but not in the index, over versions with invalid SemVer versions, to crates that were deleted from S3 for privacy-related concerns, but were still in the database.
+
+While our work here is not finished yet, we have cleaned up a lot of these inconsistencies over the past couple of months. We also started treating our database as the primary source of truth, instead of saving some metadata subset in the database and another subset in the index.
```

Comment:
> instead of saving some metadata in both the database and the index

that's not quite accurate though. we're still saving data in both places, but we now have one place that has 100% of the data and one with 30%, instead of before with 80%/30%.

I think I'd prefer keeping the current wording.